nntrainer/graph/network_graph.cpp

```diff
@@ -490,4 +496,43 @@ int NetworkGraph::setEdge() {
   return status;
 }
 
+void NetworkGraph::setBatchSize(unsigned int batch_size) {
+  for (auto const &layer_node : Sorted) {
+    layer_node.layer->setBatch(batch_size);
+  }
+}
+
+sharedConstTensors NetworkGraph::forwarding(sharedConstTensors input) {
+  for (unsigned int i = 0; i < Sorted.size() - 1; ++i) {
+    LayerNode &layer_node = Sorted[i];
+    if (layer_node.layer->getType() == LayerType::LAYER_IN) {
+      layer_node.layer->forwarding(input);
```

Comment:
why LAYER_IN not following the same semantics as other layers?

Reply:
Input needs input data. That is the layer which set the first data. If you are look at the assignMem(), it does not assign input memory buffer as well. It just take memory assigned from outside.

Reply:
I think it is better for graph has the reference to inputs and set first before first forwarding happens to unify the api, reduce (dynamic) type checking. Maybe later patch

Reply:
I think this is the place need to fix. Input Layer is not the only one takes input. Conv can takes and others too. Need to fix later PR.

Reply:
yes.